### PR TITLE
Feature/spd 1480 update

### PR DESCRIPTION
**Pull request by Occy88**
## !! Breaking Change !!

If this PR contains a breaking change describe the impact here

-

## Checklist - must complete before review.

---

### Absolute requirements

- [x] The task in the branch name has the correct Reviewer and the task is moved to Review
- [x] IDE warnings are resolved
- [x] I vow that I have not ignored stinky code smells
- [x] Test coverage script executed, and result is 100% on new/modified code, if not explain in the notes
- [x] Integration tests pass
- [x] Postman APIs are up to date
- [x] Reviewed front-end functionality

## Notes for the reviewer

Eg. branch to use in other repos or important things to be aware of when running etc.

-

### CODE REVIEW

File: seclea_ai/svc/api/collection/dataset.py
Comment (rogermilroy) [inline seclea_ai/svc/api/collection/dataset.py:41]: Better to wrap this all in a context manager that opens the file. That way it will be closed when finished with.

Comment (rogermilroy) [inline seclea_ai/svc/api/collection/dataset.py:41]: Better to wrap this all in a context manager that opens the file. That way it will be closed when finished with.

File: seclea_ai/svc/api/collection/model_state.py
Comment (rogermilroy) [inline seclea_ai/svc/api/collection/model_state.py:35]: Same for here - use a context manager.

Comment (rogermilroy) [inline seclea_ai/svc/api/collection/model_state.py:35]: Same for here - use a context manager.

File: setup.py
Comment (rogermilroy) [inline setup.py:8]: This is unnecessary.

Comment (rogermilroy) [inline setup.py:8]: This is unnecessary.

File: seclea_ai/seclea_ai.py
Comment (rogermilroy) [inline seclea_ai/seclea_ai.py:917]: This function needs to be removed and the usage updated to reflect that. This was to get around size limit in query params which is removed in this PR.

Comment (rogermilroy) [inline seclea_ai/seclea_ai.py:917]: This function needs to be removed and the usage updated to reflect that. This was to get around size limit in query params which is removed in this PR.

**Review by rogermilroy — CHANGES_REQUESTED** [review]
Seclea-utils version is also not right - needs to be on e84971 to be on the same version as portal.

File: seclea_ai/svc/api/collection/dataset.py
Comment (Occy88) [inline seclea_ai/svc/api/collection/dataset.py:38]:
```suggestion
        "metadata": (None, json.dumps(metadata)),
```

File: seclea_ai/svc/api/collection/dataset.py
Comment (Occy88) [reply seclea_ai/svc/api/collection/dataset.py:41]: I believe if you pass this to requests it's the standard method as you may need to pass multiple files, technically these should not be opened here anyway, I am not making any changes as these would constitute major refactoring as reflected by codescene.

File: setup.py
Comment (Occy88) [inline setup.py:8]:
```suggestion

```

File: seclea_ai/seclea_ai.py
Comment (Occy88) [reply seclea_ai/seclea_ai.py:917]: not touching this as it requires refactoring.

**Comment by rogermilroy** [comment]
@Occy88 yeah I'm not sure why it's duplicating all the comments - I will keep an eye out on the next review I do

**Comment by Occy88** [comment]
@rogermilroy let me know if there is anything else here, I'm running a final test.

**Comment by rogermilroy** [comment]
@Occy88 yes I have unresolved a few comments and replied. Once they are resolved or you tell me that you won't do them (and I do them instead) then it will be fine. The PR will be ready to merge once portal is merged.

**Review by rogermilroy — APPROVED** [review]
This is fine.

**Pull request merged**
